Controllers/ImageRecognizer.cs

```diff
@@ -1,36 +1,123 @@
 ﻿using System;
+using System.Collections;
+using System.Collections.Generic;
 using System.Diagnostics;
+using System.Drawing;
+using System.Drawing.Drawing2D;
+using System.Drawing.Imaging;
 using System.IO;
+using System.Linq;
 using Tesseract;
 
 namespace SmartSaver.Controllers
 {
     public class ImageRecognizer
     {
-        //Install-Package Tesseract -Version 4.1.0-beta1
         private readonly string TessDataPath = @".\assets\tessdata\best";
 
-        public void Recognize()
+        public void RecognizeTestImages()
         {
             DirectoryInfo dir = new DirectoryInfo(@"..\..\..\assets\Receipts");
-
-            foreach (var imgPath in dir.GetFiles("r1.jpg"))
+            foreach (var imgPath in dir.GetFiles("*.jpg"))
             {
-                Debug.WriteLine(imgPath);
-
-                using var ocrEngineBest = new TesseractEngine(TessDataPath, "lit", EngineMode.Default);
-                //ocrEngineBest.SetVariable("tessedit_char_whitelist", "0123456789,.");
-
-                using var img = Pix.LoadFromFile(imgPath.ToString());
-                var imgGray = img.ConvertRGBToGray();
-
-                var res = ocrEngineBest.Process(imgGray).GetText();
+                var res = Recognize(imgPath);
 
                 var destPath = imgPath.Directory.Parent.FullName + @"\receiptsTxt\" + imgPath.Name + ".txt";
                 Debug.WriteLine(destPath);
-                
+
                 File.WriteAllText(destPath, res);
             }
         }
+
+        private string Recognize(FileInfo imagePath)
+        {
+            Debug.WriteLine(imagePath);
+
+            // Original Image
+            using var originalImage = Image.FromFile(imagePath.FullName);
+            double scaleIndex = (double) originalImage.Width / originalImage.Height;
+
+            // Auto crop Image
+
+            // Scaled Image
+            var resizedImage = ResizeImage(originalImage, 500, (int)(500 / scaleIndex));
+
+            return ProccessImage(resizedImage);
+
+            // Split Image
+            // SplitImage(resizedImage);
+        }
+
+        /// <summary>
+        ///     Resize the image to the specified width and height.
+        /// </summary>
+        /// <param name="image">The image to resize.</param>
+        /// <param name="width">The width to resize to.</param>
+        /// <param name="height">The height to resize to.</param>
+        /// <returns>The resized image.</returns>
+        private Bitmap ResizeImage(Image image, int width, int height)
```

Comment:
Why you need to resize the image?

Reply:
Good question! Answer is [here](https://www.researchgate.net/publication/330956682_Object_character_recognition_in_C_using_Tesseract)
Long story short: if parsing images more than 500px wide the **confidence** of recognition engine is **0** (according to this article). Basically speeking, it does not recognize text in images : ) I will explain it tomorrow in more detailed way.